be/src/storage/lake/metadata_iterator.h

```diff
@@ -1,12 +1,44 @@
 // This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
 
 #pragma once
+#include <vector>
+
+#include "common/status.h"
+#include "common/utils.h"
+#include "fmt/format.h"
+#include "storage/lake/tablet_manager.h"
 
 namespace starrocks::lake {
 
-// TODO:
+class TabletManager;
+
+template <typename T>
 class MetadataIterator {
 public:
-};
+    explicit MetadataIterator(TabletManager* manager, std::string group, std::vector<std::string> vec)
+            : _manager(manager), _group(std::move(group)), _vec(std::move(vec)), _pos(0){};
+    bool has_next() {
+        if (_pos < _vec.size()) {
+            return true;
+        } else {
+            return false;
+        }
+    };
+    StatusOr<T> next() {
+        std::string name;
+        if (_pos < _vec.size()) {
+            name = std::move(_vec[_pos++]);
+        } else {
+            return Status::RuntimeError(fmt::format("Out of range pos {} size {}", _pos, _vec.size()));
+        }
+        return get_meta_from_tabletmanager(_group, name);
+    };
```

Comment:
Place these definitions in cpp file and use forward declaration for TabletManager in the header file.